mesh_database_client/database.py

```diff
@@ -0,0 +1,148 @@
+from __future__ import print_function
+import os.path
+from google.auth.transport.requests import Request
+from google.oauth2.credentials import Credentials
+from google_auth_oauthlib.flow import InstalledAppFlow
+from googleapiclient.discovery import build
+from googleapiclient.errors import HttpError
+from google.oauth2 import service_account
+import pandas as pd
+import os
+import googlemaps
+from dotenv import load_dotenv
+from numpy import sqrt
+
+load_dotenv()
+
+class DatabaseClient:
+
+    def __init__(self, spreadsheet_id = None):
+
+        if spreadsheet_id is None:
+            self.spreadsheet_id = os.environ.get("SPREADSHEET_ID")
+        else:
+            self.spreadsheet_id = spreadsheet_id
+
+        SCOPES = ['https://www.googleapis.com/auth/spreadsheets.readonly']
+        # created in Google Cloud admin
+        SERVICE_ACCOUNT_FILE = 'credentials.json'
+
+        signup_sheet_range = 'Form Responses 1!A:AO'
+        links_sheet_range = 'Links!A:I'
+
+        CREDS = service_account.Credentials.from_service_account_file(
+        SERVICE_ACCOUNT_FILE, scopes=SCOPES)
+
+        self.service = build('sheets', 'v4', credentials=CREDS)
+
+        # Call the Sheets API
+        self.sheet = self.service.spreadsheets()
+
+        # load all data into dataframe
+        self.signup_df = self.get_range_as_df(signup_sheet_range)
+        self.links_df = self.get_range_as_df(links_sheet_range)
+
+        self.process_signup_df()
+        self.process_links_df()
+
+        # drop last signup_df column which does not contain data
```

Comment:
Why can't we just select a smaller range above?

Reply:
Good point, this was written after I added these processing functions.